LaunchScreenDemo/WebServices/Alamofire/Controller/JeweleryViewController.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  ClothesViewController.swift
+//  LaunchScreenDemo
+//
+//  Created by Vidhi Patel on 05/07/23.
+//
+
+import UIKit
+
+protocol GetProducts {
+    func getProducts(products: [ProductRequestModel])
+}
+
+class JeweleryViewController: UIViewController {
+
+    // MARK: - IBOutlet and Vars
+    @IBOutlet private weak var tableView: UITableView!
+    var products: [ProductRequestModel] = []
+    private let viewModel: JeweleryViewModel = JeweleryViewModel()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupUI()
+    }
+    
+    private func setupUI() {
+        navigationItem.hidesBackButton = true
+        navigationItem.title = "Women's Jewelery"
+        viewModel.delegate = self
+        viewModel.getProduct()
+    }
+}
+
+// MARK: - GetProducts
+extension JeweleryViewController: GetProducts {
+    func getProducts(products: [ProductRequestModel]) {
+        self.products = products
+        DispatchQueue.main.async {
+            self.tableView.reloadData()
+        }
+    }
+}
+
+// MARK: - UITableViewDataSource
+extension JeweleryViewController: UITableViewDataSource {
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return products.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: "JeweleryTableViewCell") as? JeweleryTableViewCell else {
+            return UITableViewCell()
+        }
+        cell.configureCell(product: products[indexPath.row])
+        return cell
+    }
+}
+
+// MARK: - UITableViewDelegate
+extension JeweleryViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, trailingSwipeActionsConfigurationForRowAt indexPath: IndexPath) -> UISwipeActionsConfiguration? {
+        let deleteAction = UIContextualAction(style: .destructive, title: "Delete") { [self] (_, _, completionHandler) in
+            viewModel.deleteProduct()
+            tableView.deleteRows(at: [indexPath], with: .fade)
```

Comment:
UI should be updated after API success not after API call this statement should be in API success